scripts/rebalance_subject_split.py

```
@@ -0,0 +1,268 @@
+
+
+def target_labels_in_row(row: dict[str, Any], label_set: set[str]) -> frozenset[str]:
+    labels: set[str] = set()
+    for ent in row.get("ner", []):
+        if isinstance(ent, list) and len(ent) >= 3 and isinstance(ent[2], str) and ent[2] in label_set:
+            labels.add(ent[2])
+    return frozenset(labels)
+
+
+def deterministic_key(sample_id: str) -> str:
+    return hashlib.sha1(sample_id.encode("utf-8")).hexdigest()
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)